core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -1277,6 +1277,7 @@ class KafkaApis(val requestChannel: RequestChannel,
       sendErrorResponseMaybeThrottle(request, Errors.TRANSACTIONAL_ID_AUTHORIZATION_FAILED.exception)
     else {
       // get metadata (and create the topic if necessary)
+      authorizeClusterOperation(request, CLUSTER_ACTION)
```

Comment:
else block assumes, all the permissions have been checked and tries to get metadata(or creates topic). can we move authorizeClusterOperation check outside of the else block? It could be first thing to check when control enters into method.